src/Core/AdminConsole/Services/Implementations/EventIntegrations/README.md

```diff
@@ -0,0 +1,355 @@
+queue, etc. in one place. The `IRabbitMqService` and `IAzureServiceBusService` implement the `IEventPublisher`
+interface and therefore can also handle directly all the message publishing functionality.
+
+## Integrations and integration configurations
+
+Organizations can configure integration configurations to send events to different endpoints -- each
+handler maps to a specific integration and checks for the configuration when it receives an event.
+Currently, there are integrations / handlers for Slack and webhooks (as mentioned above).
+
+**`OrganizationIntegration`**
+
+- The top level object that enables a specific integration for the organization.
+- Includes any properties that apply to the entire integration across all events.
+
+    - For Slack, it consists of the token:
+
+      ```json
+      { "token": "xoxb-token-from-slack" }
+      ```
+
+    - For webhooks, it is `null`. However, even though there is no configuration, an organization must
+      have a webhook `OrganizationIntegration` to enable configuration via
+      `OrganizationIntegrationConfiguration`.
+
+**`OrganizationIntegrationConfiguration`**
+
+- This contains the configurations specific to each `EventType` for the integration.
+- `Configuration` contains the event-specific configuration.
+
+    - For Slack, this would contain what channel to send the message to:
+
+      ```json
+      { "channelId": "C123456" }
+      ```
+
+    - For Webhook, this is the URL the request should be sent to:
+
+      ```json
+      { "url": "https://api.example.com" }
+      ```
+
+- `Template` contains a template string that is expected to be filled in with the contents of the
+  actual event.
+    - The tokens in the string are wrapped in `#` characters. For instance, the UserId would be
+      `#UserId#`
+    - The `IntegrationTemplateProcessor` does the actual work of replacing these tokens with
+      introspected values from the provided `EventMessage`.
+    - The template does not enforce any structure -- it could be a freeform text message to send via
+      Slack, or a JSON body to send via webhook; it is simply stored and used as a string for the most
+      flexibility.
+
+**`OrganizationIntegrationConfigurationDetails`**
+
+- This is the combination of both the `OrganizationIntegration` and `OrganizationIntegrationConfiguration` into
+  a single object. The combined contents tell the integration's handler all the details needed to send to an
+  external service.
+- An array of `OrganizationIntegrationConfigurationDetails` is what the `EventIntegrationHandler` fetches from
+  the database to determine what to publish at the integration level.
+
+# Building a new integration
+
+These are all the pieces required in the process of building out a new integration. For
+clarity in naming, these assume a new integration called "Example".
+
+## IntegrationType
+Add a new type to `IntegrationType` for the new integration
```

Comment:
⛏️ Newline above and end in a period.

I think because of the above formatting issues this and later break problems are occurring.